src/citrine/informatics/executions/generative_design_execution.py

```diff
@@ -0,0 +1,111 @@
+from citrine._session import Session
+from citrine._utils.functions import format_escaped_url
+from citrine.informatics.design_candidate import GenerationExecutionResult
+
+
+class GenerativeDesignExecution(
+    Resource["GenerativeDesignExecution"], Pageable, AsynchronousObject
+):
+    """The execution of a Generative Design task.
+
+    Possible statuses are INPROGRESS, SUCCEEDED, and FAILED.
+    Design executions also have a ``status_description`` field with more information.
+
+    """
+
+    _paginator: Paginator = Paginator()
+    _collection_key = 'response'
+    _in_progress_statuses = ["INPROGRESS"]
+    _succeeded_statuses = ["SUCCEEDED"]
+    _failed_statuses = ["FAILED"]
+    _session: Optional[Session] = None
+    project_id: Optional[UUID] = None
+
+    uid: UUID = properties.UUID('id', serializable=False)
+    """:UUID: Unique identifier of the workflow execution"""
+    status = properties.Optional(properties.String(), 'status', serializable=False)
+    """:Optional[str]: short description of the execution's status"""
+    status_description = properties.Optional(
+        properties.String(), 'status_description', serializable=False)
+    """:Optional[str]: more detailed description of the execution's status"""
+    status_info = properties.Optional(
+        properties.List(properties.String()),
+        'status_info',
+        serializable=False
+    )
+    """:Optional[List[str]]: human-readable explanations of the status"""
+    created_by = properties.Optional(properties.UUID, 'created_by', serializable=False)
+    """:Optional[UUID]: id of the user who created the resource"""
+    updated_by = properties.Optional(properties.UUID, 'updated_by', serializable=False)
+    """:Optional[UUID]: id of the user who most recently updated the resource,
+    if it has been updated"""
+    create_time = properties.Optional(properties.Datetime, 'create_time', serializable=False)
+    """:Optional[datetime]: date and time at which the resource was created"""
+    update_time = properties.Optional(properties.Datetime, 'update_time', serializable=False)
+    """:Optional[datetime]: date and time at which the resource was most recently updated,
+    if it has been updated"""
+
+    def __init__(self):
+        """GenerativeDesign executions are not directly instantiated by the user."""
+        pass  # pragma: no cover
+
+    def __str__(self):
+        return '<GenerativeDesignExecution {!r}>'.format(str(self.uid))
+
+    def _path(self):
+        return format_escaped_url(
+            # TODO: Check this path once the backend is ready
+            '/projects/{project_id}/generative-design/',
+            project_id=self.project_id,
+        )
+
+    @classmethod
+    def _build_results(
+        cls, subset_collection: Iterable[dict]
+    ) -> Iterable[GenerationExecutionResult]:
+        for generation_result in subset_collection:
+            yield GenerationExecutionResult.build(generation_result)
+
+    def results(
```

Comment:
If you have a request method on a data class, it should be connected directly to that result. So `results` here only makes sense if there can be multiple results, and the listed results are only those connected to this execution.